scripts/o3de/tests/test_manifest.py

```diff
@@ -279,3 +302,54 @@ def get_gem_json_data(gem_name: str = None, gem_path: str or pathlib.Path = None
                 as load_o3de_manifest_patch:
 
             assert manifest.get_all_gems() == expected_gem_paths
+
+class TestManifestProjects:
+    @staticmethod
+    def resolve(self):
+        return self
+
+    @staticmethod
+    def samefile(self, otherFile ):
+        return self.as_posix() == otherFile.as_posix()
+
+    @pytest.mark.parametrize("project_path, project_engine_name, engines, expected_engine_path", [
+            pytest.param(pathlib.Path('C:/project1'), 
+                'engine1', 
+                {'engine1': pathlib.Path('C:/engine1'), 'engine2': pathlib.Path('D:/engine2')},
+                pathlib.Path('C:/engine1')),
+            pytest.param(pathlib.Path('C:/project1'), 
+                'engine2', 
+                {'engine1': pathlib.Path('C:/engine1'), 'engine2': pathlib.Path('D:/engine2')},
+                pathlib.Path('D:/engine2')),
+            pytest.param(pathlib.Path('C:/engine1/project1'), 
+                '', 
+                {'engine1': pathlib.Path('C:/engine1'), 'engine2': pathlib.Path('D:/engine2')},
+                pathlib.Path('C:/engine1')),
+        ]
+    )
```

Comment:
We should add a failure test case as well, for when the project doesn't have an engine registered and it isn't a subpath of engine root directory